pkg/scheduler/scheduler_test.go

```diff
@@ -177,79 +183,113 @@ func TestScheduler(t *testing.T) {
-					gotError = err
+	for _, item := range table {
+		t.Run(item.name, func(t *testing.T) {
+			testScheduler(
+				item.injectBindError,
+				item.sendPod,
+				item.algo,
+				item.expectErrorPod,
+				item.expectForgetPod,
+				item.expectAssumedPod,
+				item.expectError,
+				item.expectBind,
+				item.eventReason,
+				eventBroadcaster,
+				testNode,
+				t,
+			)
+		})
+	}
+}
+
+func testScheduler(
+	injectBindError error,
+	sendPod *v1.Pod,
+	algo algorithm.ScheduleAlgorithm,
+	expectErrorPod *v1.Pod,
+	expectForgetPod *v1.Pod,
+	expectAssumedPod *v1.Pod,
+	expectError error,
+	expectBind *v1.Binding,
+	eventReason string,
+	eventBroadcaster record.EventBroadcaster,
+	testNode v1.Node,
+	t *testing.T,
+) {
+	var gotError error
+	var gotPod *v1.Pod
+	var gotForgetPod *v1.Pod
+	var gotAssumedPod *v1.Pod
+	var gotBinding *v1.Binding
+	configurator := &FakeConfigurator{
+		Config: &Config{
+			SchedulerCache: &schedulertesting.FakeCache{
+				ForgetFunc: func(pod *v1.Pod) {
+					gotForgetPod = pod
 				},
-				NextPod: func() *v1.Pod {
-					return item.sendPod
+				AssumeFunc: func(pod *v1.Pod) {
+					gotAssumedPod = pod
 				},
-				Recorder:     eventBroadcaster.NewRecorder(legacyscheme.Scheme, v1.EventSource{Component: "scheduler"}),
-				VolumeBinder: volumebinder.NewFakeVolumeBinder(&persistentvolume.FakeVolumeBinderConfig{AllBound: true}),
 			},
-		}
+			NodeLister: schedulertesting.FakeNodeLister(
+				[]*v1.Node{&testNode},
+			),
+			Algorithm: algo,
+			GetBinder: func(pod *v1.Pod) Binder {
+				return fakeBinder{func(b *v1.Binding) error {
+					gotBinding = b
+					return injectBindError
+				}}
+			},
+			PodConditionUpdater: fakePodConditionUpdater{},
+			Error: func(p *v1.Pod, err error) {
+				gotPod = p
+				gotError = err
+			},
+			NextPod: func() *v1.Pod {
+				return sendPod
+			},
+			Recorder:     eventBroadcaster.NewRecorder(legacyscheme.Scheme, v1.EventSource{Component: "scheduler"}),
+			VolumeBinder: volumebinder.NewFakeVolumeBinder(&persistentvolume.FakeVolumeBinderConfig{AllBound: true}),
+		},
+	}
 
-		s, _ := NewFromConfigurator(configurator, nil...)
-		called := make(chan struct{})
-		events := eventBroadcaster.StartEventWatcher(func(e *v1.Event) {
-			if e, a := item.eventReason, e.Reason; e != a {
-				t.Errorf("%v: expected %v, got %v", i, e, a)
-			}
-			close(called)
-		})
-		s.scheduleOne()
-		<-called
-		if e, a := item.expectAssumedPod, gotAssumedPod; !reflect.DeepEqual(e, a) {
-			t.Errorf("%v: assumed pod: wanted %v, got %v", i, e, a)
-		}
-		if e, a := item.expectErrorPod, gotPod; !reflect.DeepEqual(e, a) {
-			t.Errorf("%v: error pod: wanted %v, got %v", i, e, a)
-		}
-		if e, a := item.expectForgetPod, gotForgetPod; !reflect.DeepEqual(e, a) {
-			t.Errorf("%v: forget pod: wanted %v, got %v", i, e, a)
-		}
-		if e, a := item.expectError, gotError; !reflect.DeepEqual(e, a) {
-			t.Errorf("%v: error: wanted %v, got %v", i, e, a)
+	s, _ := NewFromConfigurator(configurator, nil...)
```

Comment:
Another weird one. `nil...` is the same as not passing any args after configurator. (don't fix in this PR. I'm taking notes though.)